verifier.js

```diff
@@ -55,25 +73,35 @@ class VerifierManager extends ProcessManager {
 	}
 }
 
-exports.VerifierManager = VerifierManager;
-
-const PM = exports.PM = new VerifierManager({
+const PM = new VerifierManager({
 	execFile: __filename,
-	maxProcesses: global.Config ? Config.verifierprocesses : 1,
+	maxProcesses: ('Config' in global) ? Config.verifierprocesses : 1,
 	isChatBased: false,
 });
 
 if (process.send && module === process.mainModule) {
 	// This is a child process!
 
+	// @ts-ignore
 	global.Config = require('./config/config');
 
-	require('./repl').start('verifier', cmd => eval(cmd));
-
 	process.on('message', message => PM.onMessageDownstream(message));
-	process.on('disconnect', () => process.exit());
+	process.on('disconnect', () => process.exit(0));
+
+	require('./repl').start('verifier', /** @param {string} cmd */ cmd => eval(cmd));
 }
 
-exports.verify = function (data, signature) {
+/**
+ * @param {string} data
+ * @param {string} signature
+ * @return {Promise<boolean>}
+ */
+function verify(data, signature) {
 	return PM.send({data: data, sig: signature});
+}
+
+module.exports = {
```

Comment:
It may be more efficient but it breaks circular dependencies... I guess PM doesn't have external dependencies so it's not important.